torch/csrc/jit/script/compiler.cpp

```diff
@@ -1454,15 +1454,21 @@ struct to_ir {
     auto input_values = getNamedValues(applyInputs->trees(),
                                         /*maybe_unpack*/false,
                                         identity);
-    NamedValue tensor = input_values[0];
-    NamedValue dim = NamedValue(
-        loc,
-        "dim",
-        graph->insertConstant(0, loc));
+    NamedValue gatherable = input_values[0];
     NamedValue idx = input_values[1];
-
-    return emitBuiltinCall(loc, method, "select", {tensor, dim, idx}, {}, true)
-        ->asValue(loc, method);
+    if (gatherable.value->type()->kind() == TypeKind::ListType) {
+      // if it's a list, emit a regular index selection op
+      return emitBuiltinCall(
+                 loc, method, "select", {gatherable, idx}, {}, true)
+          ->asValue(loc, method);
+
+    } else {
+      // if it's a single tensor, map tensor[idx] -> tensor.select(0, idx)
```

Comment:
We never checked it's tensor in here. `getNamedValues` has `identity` as it's checker above.

Reply:
Yeah, if it's not a tensor we'll report a compiler error with a schema matching failure (same as how things behaved before this PR). I have a small preference for this over using `ensureTensor` since the error message would have more context for the user, but happy to change.

Reply:
That makes sense, thanks for the explanation.